permuta/perm.py

```diff
@@ -1952,28 +1736,15 @@ def count_occurrences_of(self, patt):
 
     occurrences = count_occurrences_of  # permpy backwards compatibility
 
-    def occurrences_in(self, patt, self_colours=None, patt_colours=None):
-        """Find all indices of occurrences of self in patt.
-
-        If the optional colours are provided, in an occurrences the colours of
-        the patterns have to match the colours of the permutation.
-
-        Args:
-            self:
-                The classical pattern whose occurrences are to be found.
-            patt: <permuta.interfaces.Patt>
-                The patt to search for occurrences in.
-            self_colours: <tuple>
-                Optional colours on each entry of the permutation.
-            patt_colours: <tuple>
-                Optional colours on each entry of the pattern.
-
-
-        Yields: <tuple> of <int>
-            The indices of the occurrences of self in patt.
-            Each yielded element l is a tuple of integer indices of the
-            pattern patt such that:
-            self == permuta.Perm.to_standard([patt[i] for i in l])
+    def occurrences_in(
+        self,
+        patt: "Patt",
+        self_colours: List[int] = None,
+        patt_colours: List[int] = None,
+    ) -> Union[Iterator[List[Tuple[int]]], Iterator[Tuple[()]]]:
```

Comment:
I think this output should only be `Iterator[Tuple[int,...]]]` as the tuple can be of any length depending on of the length of the pattern